### DISCUSSION TITLE
How to compile a mediaSource(eg: mp3,mp4) into a js file?

### DISCUSSION THREAD

**Discussion by JesuisTong** [webpack/webpack#16109] [Q&A] [answered]
## Situation
I want to use mediaSource like `mp4` **without other network request**, but I couldn't find a way to compile `mp4` source into a `js` bundle file.

Is there any `webpack loader` to do this?

Thx a lot.

## Expect
```js
// background.jsx

// We bundle the raw mp4 into js bundle
import mp4 from './background.mp4';

// So that we could use the mp4 without waiting
export const VideoBackground = () => (<video src={mp4} />)
```

**Comment by vankop** [comment, marked as answer]
no there is no loader for this ( you can write one if you want ).  
________

Also, note sure, but maybe this will work => in webpack.config:
```js
module.exports = {
  module: {
    rules: [
      { test: /\.mp4$/, type: "asset/source" }
    ]
  }
};
```

see https://webpack.js.org/guides/asset-modules/

**Reply by JesuisTong** [reply]
Thx, maybe it's unreasonable to compile raw `mp4` into `js bundle`.

I should `preload` it, but not `compile` it.
